src/editor/editorinteractive.cc

```diff
@@ -416,26 +418,38 @@ void EditorInteractive::rebuild_showhide_menu() {
 	                  g_image_cache->get("images/wui/menus/toggle_buildhelp.png"), false, "",
 	                  shortcut_string_for(KeyboardShortcut::kCommonBuildhelp));
 
+	/** TRANSLATORS: An entry in the editor's show/hide menu to toggle whether to show maximum
+	 * building spaces that will be available if all immovables (trees, rocks, etc.) are removed */
+	showhidemenu_.add(get_display_flag(dfShowMaximumBuildhelp) ? _("Hide Maximum Building Spaces") :
+                                                                _("Show Maximum Building Spaces"),
+	                  ShowHideEntry::kMaximumBuildingSpaces,
+	                  g_image_cache->get("images/wui/menus/toggle_maxbuild.png"), false,
+	                  _("Toggle whether to show maximum building spaces that will be available if "
+	                    "all immovables (trees, rocks, etc.) are removed"),
+	                  shortcut_string_for(KeyboardShortcut::kEditorShowhideMaximumBuildhelp));
+
 	/** TRANSLATORS: An entry in the editor's show/hide menu to toggle whether the map grid is shown
 	 */
-	showhidemenu_.add(draw_grid_ ? _("Hide Grid") : _("Show Grid"), ShowHideEntry::kGrid,
+	showhidemenu_.add(get_display_flag(dfShowGrid) ? _("Hide Grid") : _("Show Grid"),
+	                  ShowHideEntry::kGrid,
 	                  g_image_cache->get("images/wui/menus/menu_toggle_grid.png"), false, "",
 	                  shortcut_string_for(KeyboardShortcut::kEditorShowhideGrid));
 
-	/** TRANSLATORS: An entry in the editor's show/hide menu to toggle whether immovables (trees,
-	 * rocks etc.) are shown */
-	showhidemenu_.add(draw_immovables_ ? _("Hide Immovables") : _("Show Immovables"),
-	                  ShowHideEntry::kImmovables,
-	                  g_image_cache->get("images/wui/menus/toggle_immovables.png"), false, "",
-	                  shortcut_string_for(KeyboardShortcut::kEditorShowhideImmovables));
+	showhidemenu_.add(
+                          /** TRANSLATORS: An entry in the editor's show/hide menu to toggle whether 
+                           * immovables (trees, * rocks etc.) are shown */
+	   get_display_flag(dfShowImmovables) ? _("Hide Immovables") : _("Show Immovables"),
```

Comment:
```suggestion
	showhidemenu_.add(
	   /** TRANSLATORS: An entry in the editor's show/hide menu to toggle whether immovables
	    * (trees, * rocks etc.) are shown */
	   get_display_flag(dfShowImmovables) ? _("Hide Immovables") : _("Show Immovables"),
```
(tabs corrected too)